File: Python/面试经典150题/二分查找/35_搜索插入位置.py
from typing import List
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            mid = l + ((r - l) >> 1)
            if target == nums[mid]:
                return mid
            elif target < nums[mid]:
                r = mid - 1
            else:
                l = mid + 1
        return l

print(Solution().searchInsert([1,3,5,6], 7))

File: JS/数组/二分查找/35_搜索插入位置.js
var searchInsert = function(nums, target) {
    let mid, left = 0, right = nums.length - 1;
    while (left <= right) {
        mid = left + ((right - left) >> 1);
        if (nums[mid] > target) {
            right = mid - 1;
        } else if (nums[mid] < target) {
            left = mid + 1;
        } else {
            return mid;
        }
    }
    if(nums[mid] > target)  return mid
    return mid + 1;
};

nums = [1,3,5,6]
console.log(searchInsert(nums,7))